### PR TITLE
Pause-commands

### PR DESCRIPTION
**Pull request by ave4224**
The purpose of this branch is to allow for commands to be run on a subsystem and resume the command that was running before them upon completion.

### CODE REVIEW

File: commands/InjectedCommand.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	}
+	
+	public InjectedCommand(Command previous, String name, double timeout) {
+		super(name, timeout);
+		this.previous = previous;
+	}
+	
+	@Override
+	final protected void initialize() {
+		if (!previous.isCanceled()) {
```
Comment (c0d3rman) [inline commands/InjectedCommand.java]: This should probably be
```
previous.isRunning() || !previous.isCanceled()
```

File: commands/InjectedCommandGroup.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	
+	public InjectedCommandGroup(Command previous, String name) {
+		super(name);
+		this.previous = previous;
+	}
+	
+	@Override
+	final protected void interrupted() {
+		onInterrupted();
+		if (!previous.isRunning()) {
```
Comment (c0d3rman) [inline commands/InjectedCommandGroup.java]: Any reason this is not the same check as `InjectedCommand`?

File: commands/InjectedCommandGroup.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	}
+	
+	@Override
+	final protected void end() {
+		onEnd();
+		if (!previous.isRunning()) {
+			previous.start();
+		}
+	}
+
```
Comment (c0d3rman) [inline commands/InjectedCommandGroup.java]: Any reason you don't override `initialize` like in `InjectedCommand`?

**Comment by ave4224** [comment]
Thanks, All fixed

**Comment by ave4224** [comment]
I added some test cases in both. I don't know how effective they will be, but pretty much what it does is inject a command saying to spin in a circle.
https://github.com/RoboticsTeam4904/2016-Code/tree/power-steering
https://github.com/RoboticsTeam4904/standard/tree/power-steering

**Comment by c0d3rman** [comment]
:+1:

**Pull request merged**